tsl/test/expected/hypercore_index_btree.out

```diff
@@ -1062,3 +1062,115 @@ select * from only_nulls_null;
+select c.relname, a.amname from pg_class c
+inner join pg_am a ON (c.relam = a.oid)
+where c.oid = :'unique_chunk'::regclass;
+      relname      |  amname   
+-------------------+-----------
+ _hyper_7_19_chunk | hypercore
+(1 row)
+
+select _timescaledb_debug.is_compressed_tid(ctid), * from :unique_chunk order by time;
+ is_compressed_tid |             time             | value 
+-------------------+------------------------------+-------
+ t                 | Mon Jan 01 01:00:00 2024 PST |     1
+ f                 | Mon Jan 01 01:00:00 2024 PST |     3
+ t                 | Mon Jan 01 02:00:00 2024 PST |     2
+(3 rows)
+
+-- Unique index creation should work but fail on uniqueness check
+\set ON_ERROR_STOP 0
+create unique index time_key on uniquetable (time);
+ERROR:  could not create unique index "_hyper_7_19_chunk_time_key"
+DETAIL:  Key ("time")=(Mon Jan 01 01:00:00 2024 PST) is duplicated.
+\set ON_ERROR_STOP 1
+-- Recompress to get all values in compressed format
+select compress_chunk(:'unique_chunk');
+             compress_chunk              
+-----------------------------------------
+ _timescaledb_internal._hyper_7_19_chunk
+(1 row)
+
+-- Everything's compressed
+select _timescaledb_debug.is_compressed_tid(ctid), * from :unique_chunk order by time;
+ is_compressed_tid |             time             | value 
+-------------------+------------------------------+-------
+ t                 | Mon Jan 01 01:00:00 2024 PST |     3
+ t                 | Mon Jan 01 01:00:00 2024 PST |     1
+ t                 | Mon Jan 01 02:00:00 2024 PST |     2
+(3 rows)
+
+-- Unique index creation should still fail
+\set ON_ERROR_STOP 0
+create unique index time_key on uniquetable (time);
+ERROR:  could not create unique index "_hyper_7_19_chunk_time_key"
+DETAIL:  Key ("time")=(Mon Jan 01 01:00:00 2024 PST) is duplicated.
+\set ON_ERROR_STOP 1
+-- Delete the conflicting value and unique index creation should succeed
+delete from uniquetable where value = 3;
+select _timescaledb_debug.is_compressed_tid(ctid), * from :unique_chunk order by time;
+ is_compressed_tid |             time             | value 
+-------------------+------------------------------+-------
+ f                 | Mon Jan 01 01:00:00 2024 PST |     1
+ f                 | Mon Jan 01 02:00:00 2024 PST |     2
+(2 rows)
+
+create unique index time_key on uniquetable (time);
+explain (costs off)
+select * from uniquetable where time = '2024-01-01 01:00';
+                                    QUERY PLAN                                     
+-----------------------------------------------------------------------------------
+ Index Scan using _hyper_7_19_chunk_time_key on _hyper_7_19_chunk
+   Index Cond: ("time" = 'Mon Jan 01 01:00:00 2024 PST'::timestamp with time zone)
+(2 rows)
+
+select * from uniquetable where time = '2024-01-01 01:00';
+             time             | value 
+------------------------------+-------
+ Mon Jan 01 01:00:00 2024 PST |     1
+(1 row)
+
```

Comment:
Now that you have a unique index, you should probably have tests that tries to insert rows that violate this constraint and demonstrate that it fails.

Reply:
There's actually some overlap with these tests and old tests. We might want to reconcile.

Reply:
Since it is already covered, I do not think this is necessary for this pull request.